Sources/ForageSDK/Component/ForagePINTextField/Vault/BtPINTextField.swift

```diff
@@ -84,7 +86,7 @@ class BasisTheoryTextFieldWrapper: UIView, VaultWrapper {
             regexDigit
         ] as [Any]
         
-        let pinRegex = try! NSRegularExpression(pattern: "^\\d{4}$")
+        let pinRegex = try! NSRegularExpression(pattern: "\\d{4}$")
```

Comment:
I'm not entirely sure how this change in the regex didn't break the validation logic?? Makes me think BT's validation logic isn't working??

Reply:
I have no idea if the following note is helpful but figured I leave just in case:
- `^\\d{4}$` - has the `^`, which requires the match to happen at the start of the string.
  - `a2222` is not a match
  - `2222` is a match
- `\\d{4}$` - has no `^`, which allows the match to happen after the start of the string 
  - `a2222` is a match
  - `2222` is a match
Note that both of these regular expressions end in `$`, which means they require that the match ends at the end of the string
  - `2222a` is not a match for **either** of those regular expressions

So, if we are able to tell BT to have a max input length of 4, then both `^\\d{4}$` and `\\d{4}$` (as well as `\\d{4}`) will be equivalent since the start-of-line `^` and end-of-line `$` match characters are irrelevant since BT would already be enforcing that the sequence can be no longer than 4.